src/renderers/action-parse.renderer.ts

```diff
@@ -116,6 +113,11 @@ export class ActionParseRenderer extends Renderer {
       elem.remove()
     })
 
+    // Add CSS-only gadgets which are used on this article
+    Gadgets.getCssOnlyGadgets(articleDetail).map((gadgetId) => {
+      moduleDependencies.styleDependenciesList.push(`ext.gadget.${gadgetId}`)
```

Comment:
CSS-only gadgets should not be added directly to the normal style dependencies to ensure the correct load order. The gadgets are loaded after the modules listed in action=parse and styles included in JS modules, but before the general `site.styles` module.

The resulting HTML should look like this:
```html
<link rel="stylesheet" href="/load.php?modules=${styleDependenciesList.join('|')}&only=styles&skin=vector">
<script async="" src="/load.php?modules=startup&only=scripts&raw=1&skin=vector"></script>
<meta name="ResourceLoaderDynamicStyles" content="">
<link rel="stylesheet" href="/load.php?modules=ext.gadget.${cssGadgetsList.join(',')}&only=styles&skin=vector">
<link rel="stylesheet" href="/load.php?modules=site.styles&only=styles&skin=vector">
```

Reply:
Fixed